### ISSUE TITLE
No validation for environment.gnome3.excludePackages

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nixpkgs#24667]
The `excludePackages` option will silently ignore any packages that aren't optional packages for GNOME. This could be confusing. The same applies for the corresponding lxqt option.

**Comment by lheckemann** [comment]
no

**Comment by doronbehar** [comment]
Yea I ran into this when trying to get rid of gnome-terminal. Maybe we should just spit a warning when these packages are attempted to be excluded?

**Comment by mvnetbiz** [comment]
I think this stems from these program modules being enabled by default? 

https://github.com/NixOS/nixpkgs-channels/blob/e0759a49733dfc3aa225b8a7423c00da6e1ecb67/nixos/modules/services/x11/desktop-managers/gnome3.nix#L366-L373

Something like this would work right? Or maybe an assertion would be better suited

```nix
programs.evince.enable = mkDefault !(lib.elem pkgs.gnome3.evince config.environment.gnome3.excludePackages);
...
```

**Comment by doronbehar** [comment]
@mvnetbiz that looks good to me. I didn't test it, but I can testify that geary, which I wanted to get rid of for a while now can be get rid of with `programs.geary.enable = false`.

**Comment by mvnetbiz** [comment]
Closed by #98510
